File: Two_Pointers_Part1/Java/merge_two_sorted_arrays.java
void merge(int[] arr, int start, int mid, int end) {
    int[] temp = new int[end - start + 1];
    int i = start, j = mid + 1, k = 0;
    while (i <= mid && j <= end) {
        if (arr[i] <= arr[j]) {
            temp[k] = arr[i];
            i++;
            k++;
        } else {
            temp[k] = arr[j];
            j++;
            k++;
        }
    }
    while (i <= mid) {
        temp[k] = arr[i];
        i++;
        k++;
    }
    while (j <= end) {
        temp[k] = arr[j];
        j++;
        k++;
    }
    for (int x = start; x <= end; x++) {
        arr[x] = temp[x - start];
    }
}

File: Two_Pointers_Part1/Python/merge_two_sorted_arrays.py
def merge(arr, start, mid, end):
    temp = [0] * (end - start + 1)
    i = start
    j = mid + 1
    k = 0
    while i <= mid and j <= end:
        if arr[i] <= arr[j]:
            temp[k] = arr[i]
            i += 1
        else:
            temp[k] = arr[j]
            j += 1
        k += 1
    while i <= mid:
        temp[k] = arr[i]
        i += 1
        k += 1
    while j <= end:
        temp[k] = arr[j]
        j += 1
        k += 1
    for i in range(start, end + 1):
        arr[i] = temp[i - start]

arr = [5, 8, 2, 6, 1, 9]
merge(arr, 0, 2, 5)

print("Merged Array:")
print(arr)
